### PR TITLE
Fixed Pipe Upload: Added Metadata, Blob index tags, CPK

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: cmd/copy.go [hunk 1]
```diff
@@ -1182,9 +1182,17 @@ func (cca *cookedCopyCmdArgs) processRedirectionUpload(blobResource common.Resou
...
 	blockBlobUrl := azblob.NewBlockBlobURL(*u, p)
+	metadata, err := common.UnMarshalToCommonMetadata(cca.metadata)
+	if err != nil {
+		return err
+	}
+	blobTags := cca.blobTags
 	_, err = azblob.UploadStreamToBlockBlob(ctx, os.Stdin, blockBlobUrl, azblob.UploadStreamToBlockBlobOptions{
-		BufferSize: int(blockSize),
-		MaxBuffers: pipingUploadParallelism,
+		BufferSize:               int(blockSize),
```
Comment (adreed-msft) [inline cmd/copy.go:1191]: Should we consider the additional similar simple to support parameters, e.g. blob tier, content headers (content type, disposition, encoding, etc.), cache control?

Comment (mohsha-msft) [reply cmd/copy.go]: Ohh my bad. I don't know why I missed that

File: cmd/zc_traverser_list.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func (l *listTraverser) traverse(preprocessor objectMorpher, processor objectPro
 		}
 
 		// listTraverser will only ever execute on the source
-		if childTraverser.isDirectory(true) && !l.recursive {
+		if !l.recursive && childTraverser.isDirectory(true) {
```
Comment (mohsha-msft) [inline cmd/zc_traverser_list.go:64]: Minor perf improvement.

**Review by siminsavani-msft — COMMENTED** [review]
LGTM!

**Review by siminsavani-msft — APPROVED** [review]

**Review by zezha-msft — APPROVED** [review]

**Pull request merged**